File: java/gov/nasa/gsfc/gmsec/api/field/U32Field.java
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file U32Field.java
 */

package gov.nasa.gsfc.gmsec.api.field;

import gov.nasa.gsfc.gmsec.api.U32;
import gov.nasa.gsfc.gmsec.api.jni.field.JNIU32Field;


/**
 * Specialized Field class that can be used to retain an unsigned 32-bit value.
 */
public class U32Field extends Field
{
	/**
	 * This constructor is for internal GMSEC API use only.
	 * @param field A JNIU32Field object.
	 */
	public U32Field(JNIU32Field field)
	{
		setInternal(field);
	}


	/**
	 * Default constructor.
	 *
	 * @param name Name of the field.
	 * @param data Data to associate with the field.
	 *
	 * @throws IllegalArgumentException Thrown if the field name is null, or contains an empty string.
	 * @throws IllegalArgumentException Thrown if the data value is null.
	 */
	public U32Field(String name, U32 data) throws IllegalArgumentException
	{
		validateName(name);

		if (data == null)
		{
			throw new IllegalArgumentException("U32Field object is null");
		}

		setInternal(new JNIU32Field(name, data));
	}


	/**
	 * Copy constructor.
	 *
	 * @param other The other U32Field object to copy.
	 *
	 * @throws IllegalArgumentException Thrown if the given U32Field object is null.
	 */
	public U32Field(U32Field other) throws IllegalArgumentException
	{
		if (other == null)
		{
			throw new IllegalArgumentException("U32Field object is null");
		}

		setInternal(new JNIU32Field((JNIU32Field) Field.getInternal(other)));
	}


	/**
	 * Returns the data stored within the U32Field.
	 *
	 * @return A U32 object.
	 */
	public U32 getValue()
	{
		JNIU32Field jU32Field = (JNIU32Field) getInternal();

		return jU32Field.getValue();
	}
}

File: framework/src/gmsec4/field/U32Field.cpp
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


#include <gmsec4/field/U32Field.h>
#include <gmsec4/internal/field/InternalU32Field.h>


using namespace gmsec::api::internal;


namespace gmsec
{
namespace api
{

U32Field::U32Field(const char* name, GMSEC_U32 value)
	: m_internal(new InternalU32Field(name, value))
{
	registerChild(m_internal);
}


U32Field::U32Field(const U32Field& other)
	: m_internal(new InternalU32Field(other.getName(), other.getValue()))
{
	registerChild(m_internal);
}


U32Field::~U32Field()
{
	delete m_internal;
}


GMSEC_U32 U32Field::getValue() const
{
	return m_internal->getValue();
}


const char* U32Field::toXML() const
{
	return m_internal->toXML();
}


const char* U32Field::toJSON() const
{
	return m_internal->toJSON();
}

} //namespace api
} //namespace gmsec
